modules/food-menu/fancy-menu.tsx

```diff
@@ -65,181 +59,137 @@ const styles = StyleSheet.create({
 
-	keyExtractor = (item: MenuItem, index: number) => index.toString()
-
-	render() {
-		const {now, meals, cafeMessage, applyFilters, foodItems} = this.props
-		const {filters} = this.state
-
-		const {label: mealName, stations} = chooseMeal(meals, filters, now)
-		const anyFiltersEnabled = filters.some((f) => f.enabled)
-		const specialsFilterEnabled = this.areSpecialsFiltered(filters)
-		const groupedMenuData = this.groupMenuData({
-			stations,
-			filters,
-			applyFilters,
-			foodItems,
-		})
-
-		let message = 'No items to show.'
-		if (cafeMessage) {
-			message = cafeMessage
-		} else if (specialsFilterEnabled && stations.length === 0) {
-			message =
-				'No items to show. There may be no specials today. Try changing the filters.'
-		} else if (anyFiltersEnabled && !size(groupedMenuData)) {
-			message = 'No items to show. Try changing the filters.'
-		}
-
-		const messageView = <NoticeView style={styles.message} text={message} />
-
-		// If the requested menu has no food items, that location is closed
-		const isOpen = Object.keys(foodItems).length !== 0
-
-		const header = (
-			<FilterToolbar
-				date={now}
-				filters={filters}
-				isOpen={isOpen}
-				onPopoverDismiss={this.updateFilter}
-				title={mealName}
-			/>
-		)
-
-		return (
-			<SectionList
-				ItemSeparatorComponent={Separator}
-				ListEmptyComponent={messageView}
-				ListHeaderComponent={header}
-				contentContainerStyle={styles.contentContainer}
-				extraData={filters}
-				keyExtractor={this.keyExtractor}
-				onRefresh={this.props.onRefresh}
-				refreshing={this.props.refreshing}
-				renderItem={this.renderItem}
-				renderSectionHeader={this.renderSectionHeader}
-				sections={groupedMenuData}
-				style={styles.inner}
-				windowSize={5}
-			/>
-		)
-	}
+	const header = (
+		<FilterToolbar
+			date={now}
+			filters={filters}
+			isOpen={isOpen}
+			onPopoverDismiss={(newFilter) => {
+				let edited = filters.map((f) =>
+					f.key === newFilter.key ? newFilter : f,
+				)
+				setFilters(edited)
+			}}
+			title={mealName}
+		/>
+	)
+
+	return (
+		<SectionList
+			ItemSeparatorComponent={Separator}
+			ListEmptyComponent={messageView}
+			ListHeaderComponent={header}
+			contentContainerStyle={styles.contentContainer}
+			extraData={filters}
+			keyExtractor={(item: MenuItemType) => item.id}
+			onRefresh={props.onRefresh}
+			refreshing={props.refreshing}
+			renderItem={({item}) => {
+				return (
+					<FoodItemRow
+						badgeSpecials={!specialsFilterEnabled}
+						corIcons={menuCorIcons}
+						data={item}
+						onPress={() =>
+							navigate('MenuItemDetailView', {item, icons: menuCorIcons})
+						}
+						spacing={{left: LEFT_MARGIN}}
+					/>
+				)
+			}}
+			renderSectionHeader={(info) => {
+				const title = info.section.title
+				const note = stationsByLabel.get(title)?.note ?? ''
+
+				return (
+					<ListSectionHeader
+						spacing={{left: LEFT_MARGIN}}
+						subtitle={note}
+						title={title}
+					/>
+				)
+			}}
```

Comment:
I am fine if we are inlining these but do we have a rule of thumb as to when we pull item and header render functions out?

Reply:
I am now biased towards "always inline". 

1. I've discovered that if you inline them, TS will infer the argument types. Without that, you need something like `SectionListItem<MenuItem[], {title: string, data: MenuItem[]}>` for `renderSectionHeader`, and something else for `renderItem`. 

2. I realized that there was a surprising amount of duplicate work going on, between the render fns and the main render method. Inlining allows us to avoid that easily (although just using a function component does, too).